src/ControlSystem/TimescaleTuner.hpp

```diff
@@ -0,0 +1,69 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+
+#include "DataStructures/DataVector.hpp"
+
+/*!
+ * \ingroup ControlSystemGroup
+ * \brief Manages control system timescales
+ *
+ * The TimescaleTuner adjusts the damping timescale, \f$\tau\f$, of the control
+ * system.\n The damping timescale is restricted to the range specified
+ * by `min_timescale` and `max_timescale`,
+ * i.e. `min_timescale`\f$\le\tau\le\f$`max_timescale`
+ *
+ * The damping time is adjusted according to the following criteria:
+ *
+ * **Decrease** the timescale by a factor of `decrease_factor` if either \n
+ * 1) the error is too large: \f$|Q| >\f$ `max_threshold` \n
+ * or \n
+ * 2) the error is changing quickly: \f$|\dot{Q}|\tau >\f$ `max_threshold`,\n
+ * AND \n
+ * 3) the error is growing: \f$\dot{Q}Q > 0\f$ \n
+ * or \n
+ * 4) the expected change in \f$Q\f$ is less than half its current value:
+ * \f$|\dot{Q}|\tau < |Q|/2\f$
+ *
+ * **Increase** the timescale by a factor of `increase_factor` if \n
+ * 1) the error is sufficiently small: \f$|Q|<\f$ `min_threshold` \n
+ * AND \n
+ * 2) the expected change in \f$Q\f$ is less than the difference between the
+ * current error and the minimum error threshold:
+ * \f$|\dot{Q}|\tau < \f$ (`min_threshold` \f$-|Q|\f$)
+ */
+
+class TimescaleTuner {
+ public:
+  TimescaleTuner(DataVector initial_timescale, double max_timescale,
+                 double min_timescale, double max_threshold,
+                 double min_threshold, double increase_factor,
+                 double decrease_factor) noexcept;
+
+  TimescaleTuner(TimescaleTuner&&) noexcept = default;
+  TimescaleTuner& operator=(TimescaleTuner&&) noexcept = default;
+  TimescaleTuner(const TimescaleTuner&) = delete;
+  TimescaleTuner& operator=(const TimescaleTuner&) = delete;
+  ~TimescaleTuner() = default;
+
+  /// returns the current timescale for each component of a FunctionOfTime
+  const DataVector& current_timescale() noexcept { return timescale_; }
+  /// manually sets all timescales to a specified value, if allowable
+  void suggested_timescale_if_in_allowable_range(
+      double suggested_timescale) noexcept;
```

Comment:
Maybe "suggested_timescale_if_in_allowable_range" -> "set_timescale_if_in_allowable_range" for clarity?